git/helpers.sh

```diff
@@ -0,0 +1,59 @@
+#
+# Deletes a branch locally and remotely
+#
+function gitdelete() {
+  if [ -n "$1" ]; then
+    local_exists=`git show-ref refs/heads/$1`
+    remote_exists=`git ls-remote origin $1`
+    if [ -n "$local_exists" ]; then
+      git branch -D $1;
+    else
+      echo "could not find local branch: $1";
+    fi
+    if [ -n "$remote_exists" ]; then
+      git ps origin :$1;
+    else
+      echo "could not find remote branch: $1";
+    fi
+  fi
+}
+
+#
+# Merges a remote source branch into a remote destination branch (that defaults to `develop`)
+# and deletes the source branch locally and remotely
+#
+function gitmerge() {
+  SOURCE_BRANCH="$1"
+  DESTINATION_BRANCH="${2:-develop}"
+  echo "merging $SOURCE_BRANCH into $DESTINATION_BRANCH..."
+  if [[ $SOURCE_BRANCH == "" ]]; then
+    echo "Please provide the name of the branch to be used"
+    exit
+  fi
+  git fetch origin $SOURCE_BRANCH && \
+  git stash && \
+  git checkout $DESTINATION_BRANCH && \
+  git pull origin $DESTINATION_BRANCH && \
+  git merge origin/$SOURCE_BRANCH
```

Comment:
Does this also do the same thing and is perhaps a bit safer?
`git stash && \`
`git fetch origin $SOURCE_BRANCH && \`
`git checkout $SOURCE_BRANCH && \`
`git pull origin $SOURCE_BRANCH && \`
`git fetch origin $DESTINATION_BRANCH && \`
`git checkout $DESTINATION_BRANCH && \`
`git pull origin $DESTINATION_BRANCH && \`
`git merge $SOURCE_BRANCH`

And if so, could this even be simplified to this?
`git stash && \`
`git fetch origin && \`
`git checkout $SOURCE_BRANCH && \`
`git pull origin $SOURCE_BRANCH && \`
`git checkout $DESTINATION_BRANCH && \`
`git pull origin $DESTINATION_BRANCH && \`
`git merge $SOURCE_BRANCH`

What happen if `pull` or `merge` leads to a merge conflict? Do we need to check for `RETURN_CODE` after each `pull`?

Reply:
1- This might be just a matter of preference, but I personally like to stay away from `fetch` if the desired outcome is "merged code" (which is the goal here). 
2- The `&& \` operator prevents the next steps from getting executed if an issue is encountered at any given step, including merge conflicts.

Reply:
2 - good to know

Reply:
as below with the `gitrelease` script, we should do the stash first to avoid potential issues in the terminal

Reply:
ok